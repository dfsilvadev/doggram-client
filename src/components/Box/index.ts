import styled, { css } from "styled-components";

const Box = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    padding: 1.5rem;
    margin-top: 1.5rem;
  `}
`;

export default Box;
