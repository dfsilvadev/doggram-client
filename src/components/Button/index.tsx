import { forwardRef, ForwardRefRenderFunction } from "react";

import * as S from "./styles";

import { ButtonProps, WrapperStyleProps } from "./types";

const Button: ForwardRefRenderFunction<WrapperStyleProps, ButtonProps> = (
  {
    children,
    size = "medium",
    fullWidth = false,
    ghost = false,
    minimal = false,
    loading = false,
    icon,
    ...props
  },
  ref
) => {
  return (
    <S.Wrapper
      size={size}
      fullWidth={fullWidth}
      hasIcon={!!icon}
      minimal={minimal}
      ghost={ghost}
      ref={ref}
      {...props}
    >
      {loading ? (
        <S.Loader />
      ) : (
        <>
          {!!icon && icon}
          {!!children && <span>{children}</span>}
        </>
      )}
    </S.Wrapper>
  );
};

export default forwardRef(Button);
