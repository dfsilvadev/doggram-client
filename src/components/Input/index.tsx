import { ForwardRefRenderFunction, forwardRef } from "react";

import * as S from "./styles";

import { TextFieldProps } from "./types";

const Input: ForwardRefRenderFunction<HTMLInputElement, TextFieldProps> = (
  {
    label,
    icon,
    name,
    iconPosition = "left",
    disabled = false,
    error,
    ...props
  },
  ref
) => {
  return (
    <S.Wrapper disabled={disabled} error={!!error}>
      {!!label && <S.Label htmlFor={name}>{label}</S.Label>}
      <S.InputWrapper>
        {!!icon && <S.Icon iconPosition={iconPosition}>{icon}</S.Icon>}
        <S.Input
          type="text"
          disabled={disabled}
          name={name}
          {...(label ? { id: name } : {})}
          {...props}
          ref={ref}
        />
      </S.InputWrapper>
      {!!error && <S.ErrorMsg>{error}</S.ErrorMsg>}
    </S.Wrapper>
  );
};

export default forwardRef(Input);
