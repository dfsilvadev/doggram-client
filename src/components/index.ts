export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Button } from "./Button";
export { default as Card } from "./Card";
export { default as Container } from "./Container";
export { default as Form } from "./Form";
export { FlexComponent as Flex } from "./Flex";
export { default as Input } from "./Input";
export { default as Navbar } from "./Navbar";
export { default as SignInForm } from "./SignInForm";
export { default as SignUpForm } from "./SignUpForm";
