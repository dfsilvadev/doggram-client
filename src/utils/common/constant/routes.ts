export const ROUTES = {
  HOME: "/",
  SIGNIN: "/sign-in",
  SIGNUP: "/sign-up",
  PROFILE: {
    HOME: "/profile",
    EDIT: "/profile/edit"
  }
} as const;
