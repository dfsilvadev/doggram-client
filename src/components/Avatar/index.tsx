import { FC } from "react";
import { Link } from "react-router-dom";
import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
import { SignOut, User } from "phosphor-react";

import { ENDPOINT } from "@/utils/common/constant/endpoints";

import * as S from "./styles";

import { AvatarProps } from "./types";

const Avatar: FC<AvatarProps> = ({ ...props }) => {
  return (
    <DropdownMenu.Root>
      <DropdownMenu.Trigger asChild>
        <S.Image {...props} aria-label="avatar" />
      </DropdownMenu.Trigger>

      <DropdownMenu.Portal>
        <S.DropdownMenuContent sideOffset={5} aria-label="profile menu">
          <S.DropdownMenuItem>
            <Link to={ENDPOINT.PROFILE.HOME}>
              <User weight="bold" /> <span>Perfil</span>
            </Link>
          </S.DropdownMenuItem>

          <S.DropdownMenuSeparator />

          <S.DropdownMenuItem>
            <SignOut weight="bold" /> <span>Sign out</span>
          </S.DropdownMenuItem>

          <S.DropdownMenuArrow />
        </S.DropdownMenuContent>
      </DropdownMenu.Portal>
    </DropdownMenu.Root>
  );
};

export default Avatar;
