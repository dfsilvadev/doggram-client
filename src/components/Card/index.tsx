import { useLayoutEffect, useRef, useState } from "react";
import { ChatCentered, Heart } from "phosphor-react";

import { Flex } from "@/components";

import * as S from "./styles";

type CardProps = {
  imageUrl: string;
  likes?: number;
  comments?: number;
};

const Card = ({ imageUrl, likes = 0, comments = 0 }: CardProps) => {
  const [height, setHeight] = useState(366);
  const itemRef = useRef<HTMLDivElement>(null);

  useLayoutEffect(() => {
    const getGridWidth = () => {
      if (itemRef.current) {
        const gridColumnWidth = itemRef.current.offsetWidth;
        setHeight(gridColumnWidth);
      }
    };

    getGridWidth();
    window.addEventListener("resize", getGridWidth);

    return () => window.removeEventListener("resize", getGridWidth);
  }, []);

  return (
    <S.Item ref={itemRef} height={height}>
      <S.ItemImage src={imageUrl} />
      <S.Overlay>
        <Flex gap="1rem">
          <Flex gap="0.4rem">
            <Heart size={18} weight="bold" /> {likes}
          </Flex>

          <Flex gap="0.4rem">
            <ChatCentered size={18} weight="bold" /> {comments}
          </Flex>
        </Flex>
      </S.Overlay>
    </S.Item>
  );
};

export default Card;
