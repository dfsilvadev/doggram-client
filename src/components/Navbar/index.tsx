import { Link, NavLink } from "react-router-dom";
import { MagnifyingGlass, PlusCircle } from "phosphor-react";

import { Avatar, Button, Input } from "@/components";

import { ROUTES } from "@/utils/common/constant/routes";

import * as S from "./styles";

import logo from "@assets/images/svg/logo.svg";

const Navbar = () => {
  return (
    <S.Header>
      <S.Content>
        <S.Logo>
          <Link to={ROUTES.HOME}>
            <img
              src={logo}
              alt="logotipo petsgram: composto por um desenho vetor de um cachorro sorrindo e com uma mancha roxa com formato de triângulo em seu olho direito."
              title="logotipo petsgram: composto por um desenho vetor de um cachorro sorrindo e com uma mancha roxa com formato de triângulo em seu olho direito."
              aria-label="logotipo petsgram"
            />
          </Link>
        </S.Logo>

        <Input
          type="search"
          name="search"
          placeholder="search"
          icon={<MagnifyingGlass />}
          iconPosition="right"
          role="search"
          aria-label="search photo"
        />

        <S.Nav>
          <NavLink to={ROUTES.HOME}>
            <MagnifyingGlass size={24} weight="bold" aria-label="home link" />
          </NavLink>

          <Button
            icon={<PlusCircle size={24} weight="bold" />}
            minimal
            size="medium"
          />

          <Avatar src="https://avatars.githubusercontent.com/u/16245261?v=4" />
        </S.Nav>
      </S.Content>
    </S.Header>
  );
};

export default Navbar;
