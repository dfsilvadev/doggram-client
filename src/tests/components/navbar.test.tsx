import { BrowserRouter } from "react-router-dom";
import { screen } from "@testing-library/react";
import { renderWithTheme } from "@/utils/tests";

import { Navbar } from "@/components";

import { ROUTES } from "@/utils/common/constant/routes";

describe("Navbar", () => {
  it("render currectly", () => {
    const { container } = renderWithTheme(<Navbar />, {
      wrapper: BrowserRouter
    });

    expect(container).toMatchSnapshot();
  });

  it("render the logo", () => {
    renderWithTheme(<Navbar />, {
      wrapper: BrowserRouter
    });

    expect(
      screen.getByRole("img", { name: /logotipo petsgram/i })
    ).toBeInTheDocument();
  });

  it("render a search input", () => {
    renderWithTheme(<Navbar />, {
      wrapper: BrowserRouter
    });

    expect(
      screen.getByRole("search", { name: /search photo/i })
    ).toBeInTheDocument();
  });

  it("render a link to the home page", () => {
    renderWithTheme(<Navbar />, {
      wrapper: BrowserRouter
    });

    expect(
      screen.getByRole("link", { name: /home link/i })
    ).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /home link/i })).toHaveAttribute(
      "href",
      ROUTES.HOME
    );
  });
});
