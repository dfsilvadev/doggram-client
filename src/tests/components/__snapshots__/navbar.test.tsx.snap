// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render currectly 1`] = `
.c11 {
  background: #9164fa;
  width: 2rem;
  height: 2rem;
  border-radius: 0.5rem;
  cursor: pointer;
  overflow: hidden;
  border: 1px solid #f7f8fa;
  outline: 2px solid #494d4b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #8257e5;
  color: #FFFFFF;
  border: 0;
  border-radius: 0.5rem;
  padding: 1rem;
  height: 3.125rem;
  padding: 1rem;
  font-size: 1rem;
  background: none;
  color: #494d4b;
}

.c10:hover {
  background: #6f48c9;
}

.c10:hover {
  background: none;
}

.c10 svg {
  width: 1.5rem;
}

.c10 svg + span {
  margin-left: 0.5rem;
}

.c1 {
  width: 100%;
  max-width: 70rem;
  margin: 0 auto;
  padding: 0 1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 2rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  background: #f7f8fa;
  border-radius: 0.5rem;
  padding: 0 0.5rem;
  border: 2px solid transparent;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c5:focus-within {
  border: 2px solid #9164fa;
}

.c8 {
  color: #494d4b;
  font-family: Roboto,-apple-system,BlinkMacSystemFont,'Segoe UI',Poppins,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1rem;
  padding: 0.75rem 0;
  background: transparent;
  border: 2px solid transparent;
  outline: none;
  width: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #494d4b;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c7 > svg {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 1.4rem;
  line-height: 0;
}

.c4:focus-within .c6 {
  color: #9164fa;
}

.c0 {
  width: 100%;
  height: 5rem;
  background-color: #FFFFFF;
  border-bottom: 1px solid #f2f3f5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  width: 3rem;
  height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 img {
  width: 100%;
  -webkit-block-size: cover;
  -ms-flex-block-size: cover;
  block-size: cover;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <span
        class="c3"
      >
        <a
          href="/"
        >
          <img
            alt="logotipo petsgram: composto por um desenho vetor de um cachorro sorrindo e com uma mancha roxa com formato de triângulo em seu olho direito."
            aria-label="logotipo petsgram"
            src="SvgrURL"
            title="logotipo petsgram: composto por um desenho vetor de um cachorro sorrindo e com uma mancha roxa com formato de triângulo em seu olho direito."
          />
        </a>
      </span>
      <div
        class=""
      >
        <div
          class="c4 c5"
        >
          <div
            class="c6 c7"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="116"
                cy="116"
                fill="none"
                r="84"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="175.4"
                x2="224"
                y1="175.4"
                y2="224"
              />
            </svg>
          </div>
          <input
            aria-label="search photo"
            class="c8"
            name="search"
            placeholder="search"
            role="search"
            type="search"
          />
        </div>
      </div>
      <nav
        class=""
      >
        <div
          class="c9"
          content="stretch"
          direction="row"
          display="flex"
          gap="2rem"
          justify="flex-start"
          wrap="wrap"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <svg
              aria-label="home link"
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M152,208V160a8,8,0,0,0-8-8H112a8,8,0,0,0-8,8v48a8,8,0,0,1-8,8H48a8,8,0,0,1-8-8V115.5a8.3,8.3,0,0,1,2.6-5.9l80-72.7a8,8,0,0,1,10.8,0l80,72.7a8.3,8.3,0,0,1,2.6,5.9V208a8,8,0,0,1-8,8H160A8,8,0,0,1,152,208Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="24"
              />
            </svg>
          </a>
          <button
            class="c10"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="128"
                cy="128"
                fill="none"
                r="96"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="24"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="24"
                x1="88"
                x2="168"
                y1="128"
                y2="128"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="24"
                x1="128"
                x2="128"
                y1="88"
                y2="168"
              />
            </svg>
          </button>
          <picture
            aria-expanded="false"
            aria-haspopup="menu"
            class="c11"
            data-state="closed"
            id="radix-:r0:"
            type="button"
          />
        </div>
      </nav>
    </div>
  </header>
</div>
`;
